service/src/main/java/bio/terra/cbas/dependencies/github/GitHubClient.java

```diff
@@ -0,0 +1,71 @@
+package bio.terra.cbas.dependencies.github;
+
+import bio.terra.cbas.config.GitHubConfiguration;
+import com.google.gson.Gson;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GitHubClient {
+  private static final String BASE_URL = "https://api.github.com";
+  private final Client client;
+  private final String token;
+  private final Gson gson;
+
+  public GitHubClient(GitHubConfiguration gitHubConfiguration) {
+    this.client = ClientBuilder.newClient();
+    this.token = gitHubConfiguration.personalAccessToken();
+    this.gson = new Gson();
+  }
+
+  public RepoInfo getRepo(String organization, String repo) throws GitHubClientException {
+    WebTarget target = client.target(BASE_URL).path("/repos").path(organization).path(repo);
+    Response response =
+        target
+            .request(MediaType.APPLICATION_JSON_TYPE)
+            .header("Authorization", "Bearer: " + token)
+            .get();
+    if (response.getStatus() == 200) {
+      return gson.fromJson(response.readEntity(String.class), RepoInfo.class);
+    } else {
+      RepoError error = gson.fromJson(response.readEntity(String.class), RepoError.class);
```

Comment:
Were you able force different kinds of error messages here? i.e. is "a user has a token, but no access to the repo" different from some other sort of error message we may get, i.e. a 500?

Reply:
If the user hits this GitHub endpoint for a private repo with either (1) no token or (2) a token that lacks authorization for the repo, it responds with the same "Not Found" message, so we won't be able to distinguish your two situations from GitHub's response alone. 

If we need to be able to distinguish these, in my opinion that check should be in a level of abstraction above GitHubClient (like GitHubService or the API controller). I think it'll depend a bit on how we choose to wire in ECM, so my vote would be to wait until that happens before we deal with this, but I'm not adamant about it. If we want to get that check into this PR, we should coordinate a little (or catch me up) on where/when the ECM token will be retrieved.

Reply:
I don't think we need to distinguish the two immediately. I think a general error message touching on both possibilities is enough. I was just curious if there was a difference in error message that you saw.